### PR TITLE
Add logic to return channels to FakeBackendV2

### PR DESCRIPTION
**Pull request by nkanazawa1989**
### Summary

This PR adds a logic to FakeBackendV2 that parses the reported channel mapping and returns the pulse channels with method call, such as `.drive_channel(qubit)`. Because the corresponding V1 backend returns the channels, this is kind of a bug due to missing logic.

### CODE REVIEW

**Comment by mtreinish** [comment]
This has partial overlap with #8318 although this covers more of the optional attributes in the BackendV2 interface. @gadial are you ok with closing #8318 in favor of this? Or do you want to take @nkanazawa1989's additions into #8318 too

**Comment by nkanazawa1989** [comment]
The question is if we can instantiate `PulseBackendConfiguration` with V2. Seems like this approach is overkill to just parse the channels mapping, i.e. `PulseBackendConfiguration._parse_channels`.

**Comment by gadial** [comment]
> This has partial overlap with #8318 although this covers more of the optional attributes in the BackendV2 interface. @gadial are you ok with closing #8318 in favor of this? Or do you want to take @nkanazawa1989's additions into #8318 too

I think Naoki's PR is better since it is more focused; I intend to close #8318 .

File: qiskit/providers/fake_provider/fake_backend.py [hunk 1]
```diff
@@ -84,6 +86,36 @@ def __init__(self):
...
+            self._parse_channels(self._conf_dict["channels"])
+
+    def _parse_channels(self, channels):
+        type_map = {
+            "acquire": pulse.AcquireChannel,
+            "drive": pulse.DriveChannel,
+            "measure": pulse.MeasureChannel,
+            "control": pulse.ControlChannel,
+        }
+        identifier_pattern = re.compile(r"\D+(?P<index>\d+)")
```
Comment (mtreinish) [inline qiskit/providers/fake_provider/fake_backend.py:99]: Not that regex time here will be a big deal but do you think it'll be worthwhile to move this to a class level variable so we don't have to have the compile time on each instantiation of a `FakeBackendV2`?

**Review by mtreinish — APPROVED** [review]
LGTM, one inline comment/question but not a blocker.

**Pull request merged**